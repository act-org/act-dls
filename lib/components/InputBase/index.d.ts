/**
 * @prettier
 */
import * as React from 'react';
import { InputProps } from '@material-ui/core';
export declare type Props = InputProps;
declare const InputBase: React.FC<Props>;
export default InputBase;
//# sourceMappingURL=index.d.ts.map