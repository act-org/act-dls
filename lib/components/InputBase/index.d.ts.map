{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/InputBase/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAC/B,OAAO,EAAS,UAAU,EAAE,MAAM,mBAAmB,CAAC;AAEtD,oBAAY,KAAK,GAAG,UAAU,CAAC;AAE/B,QAAA,MAAM,SAAS,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAE9B,CAAC;AAEF,eAAe,SAAS,CAAC"}