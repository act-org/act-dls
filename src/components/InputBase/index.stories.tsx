/**
 * @prettier
 */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';

import InputBase from '.';

export const Base = (): React.ReactElement<any> => {
  const [value, setValue] = React.useState('Input');

  return (
    <InputBase
      autoFocus={boolean('Auto Focus?', false)}
      color={select(
        'Color',
        {
          Primary: 'primary',
          Secondary: 'secondary',
        },
        'primary',
      )}
      disabled={boolean('Disabled?', false)}
      disableUnderline={boolean('Disable Underline?', false)}
      multiline={boolean('Multiple Lines?', false)}
      onChange={(e: any): void => {
        action('onChange()');

        setValue(e.target.value);
      }}
      placeholder={text('Placeholder', 'Placeholder')}
      value={value}
    />
  );
};

export default {
  component: InputBase,
  parameters: {
    info: {
      text: 'This is the base variant of the Input component.',
    },
  },
  title: 'Atoms/Input',
};
