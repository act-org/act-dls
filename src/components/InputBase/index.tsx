/**
 * @prettier
 */

import * as React from 'react';
import { Input, InputProps } from '@material-ui/core';

export type Props = InputProps;

const InputBase: React.FC<Props> = (props: Props): React.ReactElement<any> => (
  <Input {...props} />
);

export default InputBase;
