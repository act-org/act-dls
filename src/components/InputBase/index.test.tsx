/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import InputBase from '.';

describe('InputBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(<InputBase onChange={noop} value="Value" />);

    expect(container).toMatchSnapshot();
  });
});
