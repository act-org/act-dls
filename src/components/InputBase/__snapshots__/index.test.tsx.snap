// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBase matches the snapshot 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline"
  >
    <input
      class="MuiInputBase-input MuiInput-input"
      type="text"
      value="Value"
    />
  </div>
</div>
`;
